docs/content/docs/concepts/Breaking Changes and Supported Patterns.md

```diff
@@ -0,0 +1,293 @@
+
+> **_NOTE:_** Another way to get some tooling support for this is to transform the types when exporting them either manually in the source or as part of publishing. For example if API-extractor could replace `@out` interfaces with types that are not constructable (extend a class with a protected member and private constructor).
+
+## Optional Function Arguments
+
+TypeScript has soundness issues with functions and optional arguments.
+
+For example:
+
+```typescript
+// Example function with an optional argument
+function processItem(item: string, override?: string): string {
+  return override ?? `processed: ${item}`;
+}
+
+function processItems(items: string[], f: (item: string) => string): string[] {
+  // When an function which requires two arguments is needed (like with Array.map, since it passes the item and the index),
+  // a function that requires fewer can be used.
+  return items.map(f);
+}
+
+// Functions with optional arguments can be assigned to types missing those arguments, as is done here.
+// When combined with allowing passing more arguments than needed, optional arguments can be passed incorrectly typed and unexpected data.
+// In this case the array ends up [1, 2, 3]: its not even an array of strings like its type requires.
+const notActuallyStrings: string[] = processItems(["a", "b", "c"], processItem);
+alert(notActuallyStrings);
+```
+
+This issue means that code that gets a function as a parameter can gain unsound behaviors if an optional argument is added to the function.
+For example in the case above if `processItem` started without the optional `override`, everything would work fine producing `["a", "b", "c"]`.
+However adding the optional `override` causes the output array to be `[1, 2, 3]` which would almost certainly violate the documented behavior from before the extra argument was added. This can be avoided by doing any of the following:
+
+1. When passing a function in a way that could implicitly convert its type, wrap it in a lambda (in the example, replace `processItem` at the end with `(s) => processItem(s)`).
+2. Avoid casting functions to have more arguments. In the example replace `(item: string) => string)` with `(item: string, index: number) => string)`
+3. Don't have optional arguments.
+4. Pick one of the above for every function type usage.
+
+The first two approaches are hard to audit/enforce but allows the package to add optional arguments to functions.
+They also make adding extra arguments to callback function types an incompatible change.
+
+This makes the most pragmatic solution to do all of:
+
+1. Try and do #2 above (avoid casting functions to have more arguments). The package must do this everywhere its relevant and technically only User Code code that follows this will be considered a supported pattern.
```

Comment:
```suggestion
1. Try and do #2 above (avoid casting functions to have more arguments). The package must do this everywhere it's relevant and technically only User Code that follows this will be considered a supported pattern.
```